lib/MusicBrainz/Server/Data/Search.pm

```diff
@@ -402,12 +402,15 @@ sub schema_fixup
                 my $area = delete $data->{$json_prop};
                 $area->{gid} = $area->{id};
                 $area->{id} = 1;
+                for my $area_prop (qw( type type-id )) {
+                    delete $area->{$area_prop};
+                }
                 $data->{$prop} = MusicBrainz::Server::Entity::Area->new($area);
             }
         }
     }
     if ($type eq 'artist' && defined $data->{gender}) {
-        $data->{gender} = MusicBrainz::Server::Entity::Gender->new( name => ucfirst($data->{gender}) );
+        $data->{gender} = MusicBrainz::Server::Entity::Gender->new( name => ucfirst($data->{gender}->{name}) );
```

Comment:
I'm not sure I understand all these `->{name}` additions because it suggests to me the output is incompatible, and the search server should be fixed instead.